Manual claims that `run` returns `nothing`

adomasbaliuka opened the pull request:
This  manual page claims that `run` returns `nothing`. However  (in version v1.5.3) it returns a `Process` object (I'm not sure if it always does).

Note: this commit does not (yet) propose changes that can be merged because I don't know for sure how to correct the error.
Note: the more recent (unreleased) versions of this file seem to have the same error.

To reproduce:
```julia
julia> VERSION
v"1.5.3"
julia> typeof(run(`echo`))
Base.Process
```

adomasbaliuka committed:
Manual claims that `run` returns `nothing`

This  manual page claims that `run` returns `nothing`. However, in version v1.5.3, it returns a `Process` object (I'm not sure if it always does).

To reproduce:
```julia
julia> VERSION
v"1.5.3"
julia> typeof(run(`echo`))
Base.Process
```

jmkuhn commented:
You could clarify the meaning by changing the order of the sentence:

If the external command fails to run successfully, the run method returns `nothing`, and throws an [`ErrorException`](@ref).

adomasbaliuka commented:
@jmkuhn Thanks to your comment, I now possibly understand the intended meaning of the sentence. However, I still disagree with the wording: a function that throws an exception cannot return any object. In particular, it does not return `nothing`, which is an object of type `Nothing`. If my understanding of this matter is correct, I believe it would be better to completely remove the statement about returning something in case of error.

Suggested change:
If the external command fails to run successfully, the run method throws an ErrorException.

adomasbaliuka committed:
Remove remark on return of `run` in case of error

The original text could be misinterpreted as saying that `run` returns `nothing`, which is an object of type `Nothing`. As far as I can tell, there is no circumstance where the method `run(cmds::Base.AbstractCmd, args...; wait) in Base at process.jl:437` (line as of v1.5.3) actually returns `nothing::Nothing`.

I deleted the remark about returning `nothing` from the manual page in question.

adomasbaliuka commented:
@jmkuhn  I updated the pull request. The proposed state of the manual file contains changes that I would agree with.

**Note: I have not proposed (and don't know the procedure for proposing) adoption of this change for the same manual file in different versions of Julia.** 

Reproduction of latest commit message (in case it is overlooked):
> The original text could be misinterpreted as saying that `run` returns `nothing`, which is an object of type `Nothing`. As far as I can tell, there is no circumstance where the method `run(cmds::Base.AbstractCmd, args...; wait) in Base at process.jl:437` (line as of v1.5.3) actually returns `nothing::Nothing`.
> 
> I deleted the remark about returning `nothing` from the manual page in question.

JeffBezanson commented:
Welcome, thanks for finding this!
